packages/query-core/src/types.ts

```diff
@@ -9,6 +9,10 @@ import type { QueryFilters, QueryTypeFilter, SkipToken } from './utils'
 import type { QueryCache } from './queryCache'
 import type { MutationCache } from './mutationCache'
 
+export type DistributiveOmit<TKey, TValue> = TKey extends any
+  ? Pick<TKey, Exclude<keyof TKey, TValue>>
+  : never
```

Comment:
it's not a big deal, but how about below naming like below?

```suggestion
export type DistributiveOmit<TObject, TKey> = TObject extends any
  ? Pick<TObject, Exclude<keyof TObject, TKey>>
  : never
```

Reply:
Yeah my bad, that's better 😂